publisher/server/publisher.go

```diff
@@ -203,12 +251,21 @@ func max(n []int) int {
 	return val
 }
 
+func (s *Server) debugF(format string, v ...interface{}) {
```

Comment:
I'm not sure the server should handle logging. We should be using a separate package for log handling like [`Logrus`](https://github.com/Sirupsen/logrus) which supports shipping messages at certain log levels. We use this library on some of our other components as well.

What do you think?